msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -3381,6 +3405,72 @@ else if(getUserId()==null&&this.forumManager.getAnonRole()==true){
     return decoTopic;
   }
 
+	public void setAnonymousManager(AnonymousManager anonymousManager)
+	{
+		this.anonymousManager = anonymousManager;
+	}
+
+	public boolean isAnonymousEnabled()
+	{
+		return anonymousManager.isAnonymousEnabled();
+	}
+
+	public boolean isSiteHasAnonymousTopics()
+	{
+		return forumManager.isSiteHasAnonymousTopics(getSiteId());
+	}
+
+	/**
+	 * Determines whether the current user should see anonymous IDs in the context of the specified topic.
+	 * @param topic
+	 * @param true if the topic is anonymous and revealIDsToRoles is disabled or
+	 * revealIDsToRoles is enabled, but the user doesn't have the permission to identify users in this topic
+	 */
+	private boolean isUseAnonymousId(Topic topic)
+	{
+		if (topic == null)
+		{
+			throw new IllegalArgumentException("isUseAnonymousId invoked with null topic");
+		}
+
+		if (!isAnonymousEnabled())
+		{
+			return false;
+		}
+
+		// if topic.postAnonymous
```

Comment:
The return statement below was hard to wrap my head around, so I added this pseudocode for clarity.
I think it would be useful for anybody who needs to modify / understand this in the future.